### PR TITLE
Set og:description meta tag to diary entry description

### LINKED ISSUE #1361: Diary posts should populate Open Graph tags for social sharing
It makes me sad when I try to share a diary post on a social site like Facebook or Twitter, and the site can't automatically fetch the title and highlight image because we don't set the necessary Open Graph tags.

Open Graph debug tool here:
https://developers.facebook.com/tools/debug/sharing/

For example, if I try to share this post:
https://www.openstreetmap.org/user/PlaneMad/diary/39846

I will end up with a generic share item with no image, and the description just talks about OpenStreetMap, not a description of the post that I am trying to share:

screenshot 2016-11-08 09 35 28

### PR DESCRIPTION
**Pull request by AntonKhorev**
Uses the first paragraph with text truncated to 500 characters as an OpenGraph description on diary entry pages. Uses the default text on other pages or if no paragraph with text is found.

Fixes #1361

### CODE REVIEW

**Review by gravitystorm — CHANGES_REQUESTED** [review]
Just a small tweak here, otherwise looks good to me.

File: app/helpers/open_graph_helper.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
 module OpenGraphHelper
   require "addressable/uri"
 
-  def opengraph_tags(title = nil, og_image = nil, og_image_alt = nil)
+  def opengraph_tags(title = nil, properties = {})
```
Comment (gravitystorm) [inline app/helpers/open_graph_helper.rb:4]: This helper is only called in one place in the codebase, and it is called with two parameters. So the defaults never apply. They can be omitted entirely. 

Even if you call it as `opengraph_tags(nil, nil)` (or things that evaluate to nil, like an undefined instance variable) then `properties` will be `nil`, not `{}`.

Comment (AntonKhorev) [reply app/helpers/open_graph_helper.rb]: removed default values

**Review by kcne — APPROVED** [review]
This is very useful and a great piece of work. The code seems fine as far as I can tell. This update also aligns well with PR addressing social sharing functionality #4985.

**Review by gravitystorm — APPROVED** [review]

**Comment by gravitystorm** [comment]
Merged, thanks!

**Pull request merged**